test_set_registers_57 fails (possibly specific to Maxwell)

<!--

Thanks for opening an issue! To help the Numba team handle your information
efficiently, please first ensure that there is no other issue present that
already describes the issue you have
(search at https://github.com/numba/numba/issues?&q=is%3Aissue).

For more general "how do I do X?" type questions, please speak to us in real
time on https://gitter.im/numba/numba or post to the Numba mailing list
https://groups.google.com/a/continuum.io/forum/#!forum/numba-users.

-->

## Reporting a bug

<!--

Before submitting a bug report please ensure that you can check off these boxes:

-->

- [X] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [X] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

<!--

Please include details of the bug here, including, if applicable, what you
expected to happen!

-->

Running the testsuite on a GTX 960 yields this failure:

```
======================================================================
FAIL: test_set_registers_57 (numba.cuda.tests.cudadrv.test_linker.TestLinker)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gmarkall/numbadev/numba/numba/cuda/cudadrv/devices.py", line 224, in _require_cuda_context
    return fn(*args, **kws)
  File "/home/gmarkall/numbadev/numba/numba/cuda/tests/cudadrv/test_linker.py", line 92, in test_set_registers_57
    self.assertEquals(57, compiled._func.get().attrs.regs)
AssertionError: 57 != 56
```

This was also reported by @stuartarchibald when discussing #4954:

> The same on @stuartarchibald's machine with a GTX 750 Ti doesn't segfault, but has the following failure:
>
> ```
> numba/cuda/tests/cudadrv/test_linker.py", line 93, in test_set_registers_57 self.assertEquals(57, compiled._func.get().attrs.regs) AssertionError: 57 != 56
> ```

So this may be an issue specific to Maxwell-architecture GPUs.

--- a/numba/cuda/tests/cudadrv/test_linker.py
+++ b/numba/cuda/tests/cudadrv/test_linker.py
@@ -86,17 +86,26 @@ class TestLinker(CUDATestCase):
         self.assertTrue(A[0] == 123 + 2 * 321)
 
     @require_context
+    def test_set_registers_no_max(self):
+        """Ensure that the jitted kernel used in the test_set_registers_* tests
+        uses more than 57 registers - this ensures that test_set_registers_*
+        are really checking that they reduced the number of registers used from
+        something greater than the maximum."""
+        compiled = cuda.jit(function_with_lots_of_registers)
+        compiled = compiled.specialize(np.empty(32), *range(6))
+        self.assertGreater(compiled._func.get().attrs.regs, 57)
+
+    @require_context
     def test_set_registers_57(self):
         compiled = cuda.jit(max_registers=57)(function_with_lots_of_registers)
         compiled = compiled.specialize(np.empty(32), *range(6))
-        self.assertEquals(57, compiled._func.get().attrs.regs)
+        self.assertLessEqual(compiled._func.get().attrs.regs, 57)
 
     @require_context
     def test_set_registers_38(self):
         compiled = cuda.jit(max_registers=38)(function_with_lots_of_registers)
         compiled = compiled.specialize(np.empty(32), *range(6))
-        self.assertEquals(38, compiled._func.get().attrs.regs)
-
+        self.assertLessEqual(compiled._func.get().attrs.regs, 38)
 
 
 if __name__ == '__main__':

